File: scripts/check_indexes.js
const { MongoClient } = require('mongodb');

const url = 'mongodb://localhost:27017';
const dbName = 'pr_analyzer';

async function main() {
  const client = new MongoClient(url);

  try {
    await client.connect();
    console.log('Connected successfully to server');

    const db = client.db(dbName);
    const collection = db.collection('pullrequests');

    const indexes = await collection.indexes();
    console.log('Indexes on pullrequests collection:');
    console.log(JSON.stringify(indexes, null, 2));

    // Check for prId index specifically
    const prIdIndex = indexes.find(index => index.key.prId);
    if (prIdIndex) {
      console.log('prId index found:');
      console.log(JSON.stringify(prIdIndex, null, 2));
    } else {
      console.log('No specific index found for prId');
    }

  } catch (err) {
    console.error('An error occurred:', err);
  } finally {
    await client.close();
  }
}

main().catch(console.error);

File: scripts/check_indexes.py
from pymongo import MongoClient
import os

# MongoDB connection string
connection_string = "mongodb+srv://mentor:mentor@cluster0.hpj3khd.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0"

try:
    # Connect to MongoDB
    client = MongoClient(connection_string)

    # Access the pr_analyzer database
    db = client["pr_analyzer"]

    # Access the pullrequests collection
    pullrequests = db["pullrequests"]

    # Get and print the indexes
    indexes = pullrequests.index_information()
    print("Indexes on the pullrequests collection:")
    for index_name, index_info in indexes.items():
        print(f"Index Name: {index_name}")
        print(f"Index Info: {index_info}")
        print("---")

    # Check for the vectorSemanticSearch index
    if "vectorSemanticSearch" in indexes:
        print("vectorSemanticSearch index exists.")
        print("vectorSemanticSearch index details:")
        print(indexes["vectorSemanticSearch"])
    else:
        print("vectorSemanticSearch index does not exist.")

except Exception as e:
    print(f"An error occurred: {e}")

finally:
    # Close the connection
    if 'client' in locals():
        client.close()
